whylog/constraints/__init__.py

```diff
@@ -146,13 +148,39 @@ class TimeConstraint(AbstractConstraint):
 
     PARAMS = sorted([MIN_DELTA, MAX_DELTA])
 
+    def __init__(self, groups=None, param_dict=None, params_checking=True):
+        super(TimeConstraint, self).__init__(groups, param_dict, params_checking)
+        param_min_delta = self.params.get(self.MIN_DELTA)
+        param_max_delta = self.params.get(self.MAX_DELTA)
+        if param_min_delta is not None and param_max_delta is not None:
+            self.verify = self._verify_both
+            self._min_delta = datetime.timedelta(seconds=param_min_delta)
+            self._max_delta = datetime.timedelta(seconds=param_max_delta)
+        elif param_max_delta is not None:
+            self.verify = self._verify_max
+            self._max_delta = datetime.timedelta(seconds=param_max_delta)
+        elif param_min_delta is not None:
+            self.verify = self._verify_min
+            self._min_delta = datetime.timedelta(seconds=param_min_delta)
+
     def _check_optional_params(self, correct_param_names, actual_param_names):
         if self.MIN_DELTA not in actual_param_names and self.MAX_DELTA not in actual_param_names:
             raise ConstructorParamsError(self.TYPE, correct_param_names, actual_param_names)
 
+    def _verify_min(self, group_contents, param_dict):
+        earlier_date, later_date = group_contents
+        return later_date - earlier_date >= self._min_delta
+
+    def _verify_max(self, group_contents, param_dict):
+        earlier_date, later_date = group_contents
+        return later_date - earlier_date <= self._max_delta
+
+    def _verify_both(self, group_contents, param_dict):
+        earlier_date, later_date = group_contents
+        return self._max_delta >= later_date - earlier_date >= self._min_delta
+
     def verify(self, group_contents, param_dict):
-        # TODO remove mock
-        return True
+        raise WrongConstraintClassSetup(self.TYPE)
```

Comment:
the only way to hit this is to not hit any `if` or `elif` blocks in the constructor. Therefore raise this exception in the `else` block there, so that the feedback about misconstructing the object is immediate rather than delayed to a point where someone calls `verify()` and the traceback is useless.